lib/rails_admin/config.rb

```diff
@@ -360,7 +360,13 @@ def reload!
       def visible_models(bindings)
         visible_models_with_bindings(bindings).sort do |a, b|
           if (weight_order = a.weight <=> b.weight) == 0
-            a.label.casecmp(b.label)
+            result = a.label.casecmp(b.label)
+            if result.zero? && a.try(:model) && b.try(:model)
+              # If two models have the same label, fallback to the model name
+              # for a deterministic ordering.
+              result = a.model.name.casecmp(b.model.name)
+            end
+            result
```

Comment:
Having something like this can be simpler?

```ruby
      def visible_models(bindings)
        visible_models_with_bindings(bindings).sort_by do |model|
          [model.weight, model.label.downcase, model.model.name.downcase]
        end
      end
```